### PR TITLE
bugfix(FDS-433):[Danger]: Run danger when base and target branch is not develop or main

### PR DESCRIPTION
**Pull request by rociobaena**
## PR Author Checklist

- [x] PR title adheres to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#specification)
- [x] I have finished reviewing the contents of this PR for accuracy
- [x] If needed, I have included a [Changeset](https://github.com/changesets/changesets) and it follows [Semantic Versioning principles](https://semver.org/)
- [x] I have completed all of the above and have enabled `auto-merge` for this PR

### CODE REVIEW

File: .danger/index.ts [hunk 1]
```diff
@@ -35,16 +35,20 @@ const isCurrentDevelopOrMain = ['main', 'develop'].includes(currentBranch);
...
 const isCurrentDevelop = currentBranch === 'develop';
+const isTargetBranchMain = targetBranch === 'main';
+
+// [FDS-444]: identify release branches and PRs
+const isCurrentAReleaseBranch = currentBranch.search(/^release/) !== -1;
+const isReleasePR = isCurrentAReleaseBranch && isTargetBranchMain;
 
 const shouldDangerCheckPR =
-  isCurrentDevelopOrMain !== isTargetBranchDevelopOrMain;
+  isCurrentDevelopOrMain && isTargetBranchDevelopOrMain;
```
Comment (lordpixel) [inline .danger/index.ts:46]: @rociobaena I think we are almost there, the AND operator is ok but we still need to negate the values of `isCurrentDevelopOrMain` and `isTargetBranchDevelopOrMain `.

Comment (lordpixel) [reply .danger/index.ts:46]:
```suggestion
  !isCurrentDevelopOrMain && !isTargetBranchDevelopOrMain;
```

Our goal is to make Danger.js skip its checks if origin branch is `Develop` and target branch is `Main`.

**Review by lordpixel — CHANGES_REQUESTED** [review]
@rociobaena I left some comments, we are almost there.

BTW, this PR should point to `develop` branch.

File: .danger/index.ts (hunk 1 above)
Comment (rociobaena) [reply .danger/index.ts:46]: Manu, thank you for reviewing this. But please wait for that, right now I'm checking the fix works for all scenarios. I would like not to have you as a review for now, but you're for default, so.. just wait for me :D

Comment (rociobaena) [reply .danger/index.ts]: @lordpixel , seems the code is working now. Please take a look again.

**Review by lordpixel — CHANGES_REQUESTED** [review]
`shouldDangerCheckPR` must be negated, otherwise we'll get the opposite behavior.

The message in L52 can also be improved by adding the other branch too.

File: .danger/index.ts [hunk 2]
```diff
@@ -35,20 +35,20 @@ const isCurrentDevelopOrMain = ['main', 'develop'].includes(currentBranch);
...
 
 const shouldDangerCheckPR =
-  isCurrentDevelopOrMain !== isTargetBranchDevelopOrMain;
+  !isCurrentDevelopOrMain && !isTargetBranchDevelopOrMain;
 
 const isDevelopComparedToMain = isCurrentDevelop && isTargetBranchMain;
 
 // skip danger if current and target branch is main or develop
-if (!shouldDangerCheckPR) {
+if (shouldDangerCheckPR) {
```
Comment (lordpixel) [inline .danger/index.ts:51]:
```suggestion
if (!shouldDangerCheckPR) {
```

File: .danger/index.ts [hunk 3]
```diff
@@ -35,20 +35,20 @@ const isCurrentDevelopOrMain = ['main', 'develop'].includes(currentBranch);
...
   message(
     `Dangerfile.js does not run when the current branch is ${currentBranch}`
```
Comment (lordpixel) [inline .danger/index.ts:53]:
```suggestion
    `Skipping Danger.js checks for this PR, as it matches '${currentBranch} -> ${targetBranch}' pattern`
```

File: .danger/index.ts (hunk 2 above)
Comment (rociobaena) [reply .danger/index.ts:51]: Manu, thank you for the eyes on this.

**Review by lordpixel — APPROVED** [review]
Thanks for fixing this @rociobaena, let's give this a try once we merge. 👍

**Pull request merged**